### PR TITLE
setup.py: Sync moban template

### PR DESCRIPTION
**Pull request by jayvdb**
**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [ ] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `corobo mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by sangamcse** [comment]
@gitmate-bot rebase

File: setup.py [hunk 1]
```diff
@@ -57,21 +69,48 @@ def run_tests(self):
...
+
+        set_python_path(setup_dir)
+
+        self.apidoc_commands = list()
+
+        self.apidoc_commands.append((
+            'sphinx-apidoc', '-f', '-o', source_docs_dir,
+            '--no-toc',
+            os.path.join(setup_dir, 'coalib')
+        ))
```
Comment (sangamcse) [inline setup.py]: Why double braces?

Comment (sangamcse) [reply setup.py]: Got it 😄

Comment (jayvdb) [reply setup.py]: the inner braces are a `tuple`.

Comment (jayvdb) [reply setup.py]: also, this is an imported template.  CI already verifies that it doesnt break somethings, but `setup.py` isnt extensively tested in CI.

If there is a problem with the template , raise a bug in the source repo.

**Comment by sangamcse** [comment]
I am not familiar with the windows setup. So I just took a look there but rest changes LGTM.

**Comment by jayvdb** [comment]
ack https://github.com/coala/coala/pull/5506/commits/cfd580a911fcf7444b37fd8119d9c41a6c779437 https://github.com/coala/coala/pull/5506/commits/73ce5960528cdeeac495f4eace3816f9fb3afba5

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**